### PR TITLE
feat: use allocUnsafe() where possible

### PR DESCRIPTION
**Pull request by tuyennhv**
**Motivation**
- allocUnsafe() is better in NodeJS environment
- there is an external memory spike issue in lodestar

**Description**
- Use `allocUnsafe` where possible

### CODE REVIEW

File: src/nonce.ts [hunk 1]
```diff
@@ -22,7 +23,7 @@ export class Nonce {
 
   constructor (n = MIN_NONCE) {
     this.n = n
-    this.bytes = new Uint8Array(12)
+    this.bytes = allocUnsafe(12)
```
Comment (mpetrunic) [inline src/nonce.ts:26]: this might be dangerous as it could already contain some data.

Comment (tuyennhv) [reply src/nonce.ts:26]: we immediately set value below so I think it's safe for this place but you're right, will need to review all other places

Comment (tuyennhv) [reply src/nonce.ts:26]: notice that in other dependencies of js-libp2p we do both `alloc()` and `allocUnsafe()` so we may want to do the same thing https://github.com/achingbrain/uint8arrays/blob/56329d16d6ca575c9638f3abb9601b8a034783b8/src/alloc.js#L8

**Comment by wemeetagain** [comment]
We should use the `alloc` and `allocUnsafe` utils already included in `uint8arrays`

**Comment by tuyennhv** [comment]
this was not proved to improve performance in lodestar, closing for now

**Pull request closed without merging**